runtime/src/main/java/org/corfudb/protocols/logprotocol/CheckpointEntry.java

```diff
@@ -0,0 +1,232 @@
+package org.corfudb.protocols.logprotocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.PooledByteBufAllocator;
+import lombok.*;
+import org.corfudb.protocols.wireprotocol.DataType;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.util.serializer.Serializers;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Object & serialization methods for in-stream checkpoint
+ * summarization of SMR object state.
+ */
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class CheckpointEntry extends LogEntry {
```

Comment:
Hm, it's CheckpointWriter that's specific to `SMRMap`.  This class doesn't have SMRMap details in it.  ... Hm, `CheckpointWriter` already has a top-level comment that specifically says SMRMaps are supported and has a TODO for general SMR objects support.